### PR TITLE
fix: When state is not existing set initial state for resource

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
This fix will reset the state for a resource when a resource has been imported from another instance that also contains the taoResourceWorkflow extension installed.

How to reproduce:
- Ensure your instance has taoResourceWorkflow installed
- Create item with content
- Export Item
- Open another environment with taoResourceWorkflow installed
- Import item
- Ensure the item has initially stated and defined in the workflow

### CODE REVIEW

File: helper/ElementDoesNotDefineResourceType.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) ${year} (original work) Open Assessment Technologies SA;
```
Comment (andreluizmachado) [inline helper/ElementDoesNotDefineResourceType.php:18]: ${year} (original work)? is it correct?

File: helper/ElementDoesNotDefineResourceType.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoResourceWorkflow\helper;
+
+use Exception;
+
+class ElementDoesNotDefineResourceType extends Exception
```
Comment (andreluizmachado) [inline helper/ElementDoesNotDefineResourceType.php:27]: Usually we put the suffix Exception on the exception class names. did we change it?

File: helper/NoInitialStateForResource.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) ${year} (original work) Open Assessment Technologies SA;
```
Comment (andreluizmachado) [inline helper/NoInitialStateForResource.php:18]: here as well

File: helper/NoInitialStateForResource.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoResourceWorkflow\helper;
+
+use Exception;
+
+class NoInitialStateForResource extends Exception
```
Comment (andreluizmachado) [inline helper/NoInitialStateForResource.php:27]: here as well

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by pnal — APPROVED** [review]
- [x]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [ ]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on playground (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Pull request merged**